### PR TITLE
[VBLOCKS-1913] Persist extra call info

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

This PR introduces `Call.getInitialConnectedTimestamp` (milliseconds since epoch)` and the native layer support for it. Also introduces `Call.State` caching on the native layer.

Minor fixes include better CI and precommit checks.

## Breakdown

- Android changes to serialize the call state and cache the initial connected timestamp.
- iOS changes for the above.
- JS changes for the above.
- Improve checks.

## Validation

- Manually tested using the test app on Android and on iOS.

## Additional Notes

N/A

### CODE REVIEW

**Review by charliesantos — CHANGES_REQUESTED** [review]
Do we need to write tests?

File: api/voice-react-native-sdk.api.md [hunk 1]
```diff
@@ -172,10 +172,11 @@ export class Call extends EventEmitter {
     // Warning: (ae-forgotten-export) The symbol "NativeCallInfo" needs to be exported by the entry point index.d.ts
     //
     // @internal
-    constructor({ uuid, customParameters, from, sid, to, isMuted, isOnHold, }: NativeCallInfo);
+    constructor({ uuid, customParameters, from, sid, state, to, isMuted, isOnHold, initialConnectedTimestamp, }: NativeCallInfo);
     disconnect(): Promise<void>;
     getCustomParameters(): CustomParameters;
     getFrom(): string | undefined;
+    getInitialConnectedTimestamp(): number | undefined;
```
Comment (charliesantos) [inline api/voice-react-native-sdk.api.md:179]: This is a new public API right? This is not too big of an example, but I think we need to get into the habit of following our process for any new public APIs. If the changes requires new/updated API:
- Create/update IDL and get it reviewed, before implementing into the SDK
- Check other voice SDKs answer questions like
  - Is this applicable to other voice sdks?
  - Do we need to add similar feature?
  - How is this done on other voice sdks, if applicable?
- Probably not applicable in this PR but also part of the process. If the new feature or API is complex enough
  - Create a design doc
  - Review with the team

**Comment by mhuynh5757** [comment]
Good catch regarding tests @charliesantos, they slipped my mind. The coverage is back to (almost) 100. Please take another look.

**Review by charliesantos — APPROVED** [review]

**Review by afalls-twilio — APPROVED** [review]

**Review by kpchoy — APPROVED** [review]

File: ios/TwilioVoiceReactNative+CallKit.m [hunk 2]
```diff
@@ -318,6 +318,11 @@ - (void)callDidConnect:(TVOCall *)call {
         self.callKitCompletionCallback(YES);
         self.callKitCompletionCallback = nil;
     }
+    
+    NSTimeInterval timeStamp = [[NSDate date] timeIntervalSince1970];
+    // NSTimeInterval is defined as double
+    NSNumber *timeStampObj = [NSNumber numberWithDouble: timeStamp];
```
Comment (bobiechen-twilio) [inline ios/TwilioVoiceReactNative+CallKit.m:324]: Remove space between `numberWithDouble:` and `timeStamp`

**Review by bobiechen-twilio — APPROVED** [review]
A small suggestion but feel free to merge after removing the extra space.

**Pull request merged**